hack/netpol/pkg/main/main.go

```diff
@@ -0,0 +1,679 @@
+package main
+
+import (
+	"fmt"
+	. "github.com/vmware-tanzu/antrea/hack/netpol/pkg/utils"
+	log "github.com/sirupsen/logrus"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type NetPolConfig struct {
+	pods       []string
+	namespaces []string
+	k8s        *Kubernetes
+}
+
+// common for all tests.  these get hardcoded into the Expect() clauses,
+// so, we cant easily parameterize them (well, we could, but that would
+// make the code harder to interpret.
+var pods []string
+var namespaces []string
+var p80 int
+var p81 int
+var allPods []Pod
+
+func init() {
+	p80 = 80
+	p81 = 81
+	pods = []string{"a", "b", "c"}
+	namespaces = []string{"x", "y", "z"}
+
+	for _, podName := range pods {
+		for _, ns := range namespaces {
+			allPods = append(allPods, NewPod(ns, podName))
+		}
+	}
+}
+
+func bootstrap(k8s *Kubernetes) {
+	k8s.CleanNetworkPolicies([]string{"x","y","z"})
+	//p81 := 81
+	for _, ns := range namespaces {
+		k8s.CreateOrUpdateNamespace(ns, map[string]string{"ns": ns})
+		for _, pod := range pods {
+			fmt.Println(ns)
+			k8s.CreateOrUpdateDeployment(ns, ns+pod, 1,
+				map[string]string{
+					"pod": pod,
+				})
+		}
+	}
+	k8s.CleanNetworkPolicies([]string{"x","y","z"})
+}
+
+func validate(k8s *Kubernetes, reachability *Reachability, port int) {
+	// better as metrics, obviously, this is only for POC.
+	for _, pod1 := range allPods {
+		for _, pod2 := range allPods {
+			log.Infof("Probing: %s, %s", string(pod1), string(pod2))
+			connected, err := k8s.Probe(pod1.Namespace(), pod1.PodName(), pod2.Namespace(), pod2.PodName(), port)
```

Comment:
I would think that you could probably do even better in terms of test completion time by running the probes in parallel in different goroutines.

Reply:
@sedefsavas ^ had this thought as well.  I did a quick prototype and got hung up on client-go concurrency issues, but would be happy to revisit this in the future again?

Reply:
Sounds good, I see that you opened an issue